### DISCUSSION TITLE
Is it planned that we can use vcpkg artifacts without the shell wrapper?

### DISCUSSION THREAD

**Discussion by photex** [microsoft/vcpkg#35290] [General]
Heyo!

Happy thanksgiving for those that celebrate! 🦃 

Recently I tried to start using vcpkg artifacts (`vcpkg activate`) with an existing repository which uses vcpkg in manifest mode as a submodule. This didn't work!
What I eventually realized after reading the blog post more clearly is that `vcpkg-init.cmd` played a critical part in providing a wrapper around vcpkg which will modify the environment and so on. Makes sense.

BUT 

That does present a certain adoption friction over our existing submodule (and fetch content) approach.

What's the plan here? What are we doing? Having vcpkg as a submodule is really incredible and our cmake presets accomplish a great deal as a result. Currently our toolchains for embedded targets (and some associated tools for flashing hardware) get airlifted in via *magic*, but it would appear that vckpg artifacts are the more forward facing or long term approach to this issue.

In an ideal world, once a build is configured, it wouldn't be dependent on someone running `vcpkg-init.cmd` first especially when I consider how to integrate this into developer workflows with vscode or clion etc. Currently we lean heavily on cmake presets to setup the environment based on apriori knowledge of what's going to show up before the first call to `project()` in our build.

What would really be the sauce is that some mechanism in the vcpkg toolchain would also initialize the environment based on artifact dependencies.

**Comment by AugP** [comment]
I agree that this area could be improved. Our artifacts experience remains experimental today, and unfortunately, we don't currently have the capacity to address this, but I hope we can revisit it soon.

**Reply by photex** [reply]
Conan and Conda both have similar facilities and Arm spends a lot of time in their documentation recommending vcpkg artifacts in combination with vs code. 
And if you search for "vcpkg artifacts" you even find blog posts explaining that Visual Studio classic supports it. 
I'm surprised to hear that there aren't any resources allocated to developing it! :) All indicators point to "this is the future". 
In either case, thanks for clarifying.
